### PR TITLE
Fix a bug where BrowserClient was listed as requiring Flutter

### PR DESCRIPTION
**Pull request by brianquinlan**
- In the `Client` implementations table, add "Dart" to the list of SDKs supported by `BrowserClient`

---

- [X] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/http/README.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ Some well supported implementations are:
 | Implementation | Supported Platforms | SDK | Caching | HTTP3/QUIC | Platform Native | 
 | -------------- | ------------------- | ----| ------- | ---------- | --------------- |
 | `package:http` — [`IOClient`][ioclient] | Android, iOS, Linux, macOS, Windows | Dart, Flutter | ❌ | ❌ | ❌ |
-| `package:http` — [`BrowserClient`][browserclient] | Web | Flutter | ― | ✅︎ | ✅︎ | Dart, Flutter |
+| `package:http` — [`BrowserClient`][browserclient] | Web | Dart, Flutter | ― | ✅︎ | ✅︎ | Dart, Flutter |
```
Comment (natebosch) [inline pkgs/http/README.md:127]: The `BrowserClient` doesn't run on the Dart VM, it is Web and flutter-web only. I'm not 100% sure, but I think "Dart" here refers to the VM or AOT compiles?

Comment (brianquinlan) [reply pkgs/http/README.md:127]: I was trying to indicate whether flutter is required or not. I was copying the notation used by pub e.g. 
image

That's for `package:fetch_client` - which is web-only

Comment (brianquinlan) [reply pkgs/http/README.md:127]: I do have a separate column for the platform, which is "web" in the case of `BrowserClient`

Comment (natebosch) [reply pkgs/http/README.md:127]: Ah I see, that makes sense.

**Review by natebosch — APPROVED** [review]

**Pull request merged**